### PR TITLE
[MRG][ENH] Make reject log plot prettier

### PR DESCRIPTION
**Pull request by alexrockhill**
Related to #205.

Change the coloring to be more intuitive (green=good, red=bad) and add red boxes around rejected epochs.

### CODE REVIEW

**Comment by jasmainak** [comment]
On the website I have:

image

on this PR, I have:

image

I think I prefer the former ...

**Comment by alexrockhill** [comment]
> On the website I have:
> 
> image
> 
> on this PR, I have:
> 
> image
> 
> I think I prefer the former ...

Look how it turns out on the full PR, much better. Having too many channels can be solved by `fig.gca().set_xticks(range(0, len(epochs.ch_names), 10)` for instance, that's much better than the arbitrary amount used in the first plot.

**Comment by alexrockhill** [comment]
See this artifact for how it would look with the example changed to match https://771-59485140-gh.circle-artifacts.com/0/html/auto_examples/plot_example_workflow.html#sphx-glr-auto-examples-plot-example-workflow-py

**Comment by jasmainak** [comment]
@alexrockhill better! But now the legend does not match the colors:

image

also "interpolated" is very hard to see

**Comment by jasmainak** [comment]
I would make the green even lighter ... the focus should be on the bad and interpolated channels

**Comment by dengemann** [comment]
I like the idea. Perhaps legend labels can be drawn smaller and arranged horizontally to leave more room for actual data (information / ink ratio).

**Comment by jasmainak** [comment]
Looks good now! +1 for MRG after moving the legend to the bottom as @dengemann suggested

**Comment by jasmainak** [comment]
@alexrockhill do you want to update `whats_new.rst` ?

File: autoreject/autoreject.py [hunk 1]
```diff
@@ -1203,8 +1203,8 @@ def __init__(self, bad_epochs, labels, ch_names):
         assert len(bad_epochs) == labels.shape[0]
         assert len(ch_names) == labels.shape[1]
 
-    def plot(self, orientation='vertical', show=True):
-        """Plot.
+    def plot(self, orientation='horizontal', show=True):
```
Comment (jasmainak) [inline autoreject/autoreject.py:1206]: what's the reason for changing this default?

**Comment by hoechenberger** [comment]
IMHO the green is superfluous and only makes it more difficult to actually spot the bad and interpolated segments. Also from an accessibility perspective, choosing red and green is probably not the best idea.

I’d therefore propose to simply drop the green

**Comment by alexrockhill** [comment]
I liked the dark green because it really made the red and yellow pop but no color for good is fine too.

**Comment by jasmainak** [comment]
But now the problem is that, we don't see which channels are part of the `picks`. Earlier we showed all channels and channels that were not part of the picks were shown in "no color" (gray)

image

**Comment by hoechenberger** [comment]
Oh. Ok. Well I don't insist on the change I proposed. Just wanted to point out that there might be an accessibility problem there…

**Comment by agramfort** [comment]
I find the interpolated not enough visible 😅😌😌

>

**Comment by hoechenberger** [comment]
> I find the interpolated not enough visible 😅😌😌

Same!

**Comment by jasmainak** [comment]
let me do a pass

**Comment by jasmainak** [comment]
I just pushed a commit. How do folks like this:

image

I'm going to add a `show_names` argument a la [`evoked.plot_image`](https://mne.tools/stable/generated/mne.Evoked.html#mne.Evoked.plot_image) so folks can control how dense the tick labels are.

@alexrockhill can you try on your datasets and share if they look fine on them?

**Comment by agramfort** [comment]
still a bit to red/green. See test on colorblind vision

Screenshot 2021-05-04 at 15 10 40
Screenshot 2021-05-04 at 15 10 32

simulated at https://www.color-blindness.com/coblis-color-blindness-simulator/

**Comment by jasmainak** [comment]
colortest

image

it seems to survive colorblind tests if I make it blue. Would this be okay?

**Comment by jasmainak** [comment]
tests will pass after #211 is fixed

**Comment by alexrockhill** [comment]
https://www.nature.com/articles/nmeth.1618

Maybe vermillion=bad, yellow=interpolated, bluish green=good. That's from looking at figure 2 and picking from there.

**Comment by agramfort** [comment]
let's go for it.

I think it's a real progress anyway.

sorry to nitpick...

>

**Comment by sappelhoff** [comment]
> tests will pass after #211 is fixed

just merged it - @alexrockhill you can rebase :)

**Comment by jasmainak** [comment]
@alexrockhill the rebase lost my commits? :( I'm going to rebase with what was latest ...

**Comment by jasmainak** [comment]
okay so the Nature color combo doesn't really work @alexrockhill (aside: the paper says RGB colors are from 1-255, it should be 0-255)

image

it's too crowded. I am happy with the current selection as it survived the [colorblind tests](https://github.com/autoreject/autoreject/pull/209#issuecomment-831971883). Good to merge from my end

**Comment by jasmainak** [comment]
I also tried making the green lighter, that doesn't work either:

image

Blue is better than yellow IMO

**Comment by jasmainak** [comment]
@hoechenberger any clue what the CircleCI error is about? It's complaining about the `openneuro` package

**Comment by hoechenberger** [comment]
It appears OpenNeuro has finally deisabked the previously deprecated API that `openneuro-py` is using. I have just started working on updating it, but I won't finish this today. Should have it ready by the end of the weekend though!

**Comment by jasmainak** [comment]
alright! I can wait :)

**Comment by hoechenberger** [comment]
@jasmainak OpenNeuro PR: https://github.com/hoechenberger/openneuro-py/pull/38

**Comment by hoechenberger** [comment]
@jasmainak Using `openneuro-py:main` should solve the issue for you.
If it does work, I'll try to make a release soon.

**Comment by jasmainak** [comment]
Thanks a lot @hoechenberger . Fingers crossed the CIs come green!

**Comment by hoechenberger** [comment]
@jasmainak You don't want the URL in setup.py, this should go to a concrete environment / requirements definition :)

**Comment by hoechenberger** [comment]
I'll make a release shortly so you don't have to worry about this.

**Comment by hoechenberger** [comment]
@jasmainak `openneuro-py 2021.3` is on PyPI. You should probably add this version constraint to `setup.py`

**Comment by jasmainak** [comment]
Thanks for the release! Actually, it still wasn't working when I used the URL with the development version. Let's see if it works with the release version ...

**Comment by hoechenberger** [comment]
@jasmainak This is a server timeout, that should be unrelated to `openneuro-py` itself. I tried this one locally and I can replicate the issue. I will try again in a few hours and if it persists, I'm afraid we'll have to get in touch with OpenNeuro folks

**Comment by hoechenberger** [comment]
It appears we're hitting an implicit and new concurrency limit.

A workaround is to limit the number of parallel downloads to 1:

in `examples/plot_visualize_bad_epochs.py`, change 

```python
openneuro.download(dataset=dataset, target_dir=target_dir,
                   include=[f'sub-{subject_id}/ses-meg/'])
```

to

```python
openneuro.download(dataset=dataset, target_dir=target_dir,
                   include=[f'sub-{subject_id}/ses-meg/'],
                   max_concurrent_downloads=1)
```

**Comment by hoechenberger** [comment]
@jasmainak NB for the above to work, you'll need to use `openneuro-py:main` again, sorry :(

**Comment by hoechenberger** [comment]
@jasmainak I've released `openneuro-py` 2021.4, so you don't have to use `main`.

**Comment by hoechenberger** [comment]
@jasmainak I've published `openneuro-py` 2021.5 that should be even more robust in dealing with OpenNeuro's flaky servers.

**Comment by jasmainak** [comment]
@hoechenberger apologies for the slow responses here. Autoreject maintenance is happening mostly during my free time. I've pushed an update to use the latest `openneuro`. Fingers crossed!

**Comment by hoechenberger** [comment]
Aww come on… Still failing. I'll have a look.

**Comment by hoechenberger** [comment]
It just says that the process got killed. Could that be a memory issue? `openneuro-py` is supposed to stream the data, writing chunks to disk as they are retrieved, so it won't fill up memory. But I've never profiled actual memory usage. I wonder if I need to do that now … WDYT, @agramfort?

**Comment by hoechenberger** [comment]
But actually, we're performing the exact same download in `openneuro-py`'s CI and it works without issues there:

https://github.com/hoechenberger/openneuro-py/blob/96c803dca57017b35d8b46ea2c9d2fbc52c35901/openneuro/tests/test_download.py#L9-L10

Then again this is run on GH Actions and not Circle

**Comment by hoechenberger** [comment]
@jasmainak If you're okay with temporarily granting me the required access rights on this repo, I can try to get to the bottom of this myself, without bothering you :)

**Comment by jasmainak** [comment]
@hoechenberger I just gave you write access!

**Comment by jasmainak** [comment]
you can take over this PR

**Comment by hoechenberger** [comment]
Thanks! Will look into this tomorrow!

**Comment by hoechenberger** [comment]
It seems I cannot control Circle CI (I might need to SSH into the CI system), I'm not sure which permissions would be required for this..?

**Comment by jasmainak** [comment]
I bumped your privilege to "Maintain" ... hope that does the trick :)

**Comment by hoechenberger** [comment]
Whoops.

Screen Shot 2021-05-23 at 20 37 42

**Comment by hoechenberger** [comment]
Did some small optimizations:

Screen Shot 2021-05-23 at 22 01 17

Reduced RAM consumption by 2 orders of magnitude.

Will push this to `openneuro-py` tomorrow.

**Comment by jasmainak** [comment]
wow!! that's awesome. Autoreject is pushing the boundaries of openneuro-py ;-)

File: setup.py [hunk 2]
```diff
@@ -77,7 +77,7 @@
                   'numpydoc',
                   'cython',
                   'pillow',
-                  'openneuro-py'
+                  'openneuro-py >= 2021.5'
```
Comment (hoechenberger) [inline setup.py:80]:
```suggestion
                  'openneuro-py >= 2021.7'
```

**Comment by hoechenberger** [comment]
I didn't squash this one before merging, ouch. How should we go about this?

**Comment by jasmainak** [comment]
all good. I tend not to squash in younger projects so everyone gets due credit. But disabled the merge commit option now so as to keep linear history

**Comment by jasmainak** [comment]
Thanks for the fixes!

**Pull request merged**